core/query/evalMixingQuery.ml

```diff
@@ -92,6 +150,14 @@ and base_exp = function
     end
 | QL.Apply (QL.Primitive "Empty", [v]) -> S.Empty (sql_of_query S.All v)
 | QL.Apply (QL.Primitive "length", [v]) -> S.Length (sql_of_query S.All v)
```

Comment:
The thought occurred to me. We can try that, with the caveat that it could break code that is presently working (including the non-mixing normalizer if we completely remove Sql.length)